enhancements/plain-k8s-bundles.md

```diff
@@ -0,0 +1,220 @@
+---
+title: plain-k8s-bundles
+authors:
+  - "@exdx"
+  - "@njhale"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-03-09
+last-updated: 2020-03-09
+status: implementable
+see-also:
+  - "https://docs.google.com/document/d/1kDDVVdjSjBAP3-4QnIPb67QfcR2-LEWRbvKFUDzEwrc/edit#heading=h.70izt81ko44h"
+---
+
+# plain-k8s-bundles
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Add support for a new mediatype, `k8s+v1`, to the [Operator Bundle spec](https://github.com/openshift/enhancements/blob/master/enhancements/olm/operator-bundle.md), [operator-registry](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/api/apis/operators/v1alpha1/clusterserviceversion_types.go), and [OLM](https://github.com/operator-framework/operator-lifecycle-manager) that enables packaging an operator without defining a [ClusterServiceVersion](https://github.com/operator-framework/olm-book).
+
+## Motivation
+
+Operator authors find CSVs difficult to write and maintain. Expressing an OLM-managed operator as a set of related kubernetes objects is more intuitive to users. Supporting plain kubernetes manifests makes OLM more approachable to the wider community, which is relevant in light of the Operator Framework attempting to become a CNCF project. This feature is part of the overall goal of making OLM simpler to use.
+
+Furthermore, this feature is part of the overall goal of making OLM more flexible by supporting more media types. The more media types OLM supports the more popular and accesible OLM becomes.
+
+### Goals
+
+- Support, install, build, and lifecycle `k8s+v1` bundles  
+- Support adding `k8s+v1` bundles to an [Operator Index](https://github.com/openshift/enhancements/blob/master/enhancements/olm/operator-registry.md#summary)
+- Support packageserver surfacing data on plain manifest bundles
```

Comment:
This should include the ability to also include non-managed Kubernetes manifest in the bundle that will simply be applied. Example: `ConfigMap`, `Secrets`, `Namespace` or `PersistentVolume`

Reply:
Making bundles accept arbitrary resources is distinct from CSVless bundles. I think it's something that can apply to both `registry+v1` and `k8s+v1` bundles. For this enhancement, can we focus on implementing a way to remove the CSV, and then produce follow-ups for arbitrary resources? We need to follow up on this enhancement anyway to remove CSVs as an implementation detail so we can open up more install strategies as well (i.e. not just Deployments).